app/models/initiative.rb

```diff
@@ -48,6 +50,16 @@ class Initiative < ApplicationRecord
   validates :title, presence: true, uniqueness: true
   validate :must_have_manager
 
+  def source=(_)
+    raise USE_TRANSFER_INITIATIVE_MESSAGE unless caller[0].match(/transfer_to|activemodel/)
+
+    super
+  end
+
+  def transfer_to(new_source)
+    self.source = new_source
+  end
```

Comment:
Why? using caller to determine who is calling a method is weird. Also I don't understand why you don't want people to use `source=` whats the problem with using that?